daml-lf/interpreter/src/main/scala/com/digitalasset/daml/lf/ledger/BlindingTransaction.scala

```diff
@@ -81,7 +81,9 @@ object BlindingTransaction {
 
         case ex: NodeExercises.WithTxValue[NodeId, ContractId] =>
           val state1 =
-            state.divulgeCoidTo(parentExerciseWitnesses -- ex.stakeholders, ex.targetCoid)
+            state.divulgeCoidTo(
+              (parentExerciseWitnesses union ex.choiceObservers) -- ex.stakeholders,
```

Comment:
What makes choice observers special here? Shouldn’t the node itself be disclosed via `discloseNode` above just like for observers on a consuming choice?

Reply:
The node is already disclosed above: choice observers are _informees_ of a node.
But in addition, the contract-ids must be divulged.

Reply:
Oh because choice observers are not stakeholders so they might not know the contract. Makes sense, thanks!